editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search.tsx

```diff
@@ -1,132 +1,283 @@
+          <ListItem key={`external-portal-card-${name}`} disablePadding sx={{ mb: 2 }}>
+            <ExternalPortalCard onClick={() => navigate("../" + href)}>
+              <Typography
+                variant="body2"
+                fontWeight={FONT_WEIGHT_SEMI_BOLD}
+                mr={0.5}
+                whiteSpace="nowrap"
+              >
+                External portal •
+              </Typography>
+              <Typography
+                variant="body2"
+              >
+                {href.replaceAll("/", " / ")}
+              </Typography>
+            </ExternalPortalCard>
+          </ListItem>
+        ))}
+      </List>
     </Box>
   );
 };
 
+interface SearchNodes {
+  input: string;
+  facets: ["data.fn", "data.val"],
+}
+
 const Search: React.FC = () => {
-  const [flowId] = useStore((state) => [state.id]);
-  const [query, setQuery] = useState("");
-  const [debouncedQuery, setDebouncedQuery] = useState(query);
+  const [orderedFlow, setOrderedFlow] = useStore((state) => [
+    state.orderedFlow,
+    state.setOrderedFlow,
+  ]);
 
-  const debounceSearch = useCallback(
-    debounce((input) => setDebouncedQuery(input), 500),
-    [],
-  );
+  useEffect(() => {
+    if (!orderedFlow) setOrderedFlow();
+  }, [setOrderedFlow]);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
-    const input = e.target.value;
-    setQuery(input);
-    debounceSearch(input);
-  };
+  const formik = useFormik<SearchNodes>({
+    initialValues: { input: "", facets: ["data.fn", "data.val"] },
+    onSubmit: ({ input }) => { search(input) },
+  });
 
-  const fetcher = (url: string) => fetch(url).then((r) => r.json());
-  const endpoint = `${process.env.REACT_APP_API_URL}/flows/${flowId}/search`;
-  const { data, error } = useSWR<SearchResult[]>(
-    debouncedQuery ? `${endpoint}?find=${debouncedQuery}` : null,
-    fetcher,
-  );
+  const { results, search } = useSearch({
+    list: orderedFlow || [],
+    keys: formik.values.facets
+  });
 
   return (
     <Container component={Box} p={3}>
-      <InputLabel label="Type to search" htmlFor="search">
-        <Input name="search" onChange={handleChange} id="boundaryUrl" />
-      </InputLabel>
-      <ChecklistItem
-        label="Search only data fields"
-        id={"search-data-field-facet"}
-        checked
-        inputProps={{
-          disabled: true,
-        }}
-        onChange={() => {}}
-      />
-      <Box pt={3}>
-        {error && "Something went wrong"}
-        {mockData ? <SearchResults results={mockData} /> : "Loading..."}
-      </Box>
+      <form onSubmit={formik.handleSubmit}>
+        <Typography
+          component={"label"}
+          htmlFor="search"
+          variant="h3"
+          mb={1}
+          display={"block"}
+        >
+          Search this flow and internal portals
+        </Typography>
+        <Input
+          name="search"
+          value={formik.values.input}
+          onChange={(e) => {
```

Comment:
What's the reason for switching from a handleChange to using a handleSubmit() within an onChange? 

_Is this to ensure the form is submitted with all form data for the search on each change to the input field?_

Reply:
That's correct - it's just to make it semantically clear what's happening here. When the input value changes, we're submitting the form.

This was originally called `handleChange` in an earlier commit as I hadn't set up formik and this wasn't a HTML `form` element, and it was just a function being called when the input changed.